internal/proxy/proxy_darwin.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build darwin
+//+build darwin
+
+package proxy
+
+import (
+	"errors"
+	"math/rand"
+	"net"
+	"net/http"
+	"net/http/httputil"
+	"sync"
+	"time"
+
+	"github.com/ServiceWeaver/weaver/internal/logtype"
+)
+
+// Proxy is an HTTP proxy that forwards traffic to a set of backends.
+type Proxy struct {
+	logger   logtype.Logger        // logger
+	reverse  httputil.ReverseProxy // underlying proxy
+	mu       sync.Mutex            // guards backends
+	backends []string              // backend addresses
+}
+
+// NewProxy returns a new proxy.
+func NewProxy(logger logtype.Logger) *Proxy {
+	p := &Proxy{logger: logger}
+	p.reverse = httputil.ReverseProxy{
+		Director: p.director,
+		Transport: &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+			DialContext: (&net.Dialer{
+				Timeout:   30 * time.Second,
+				KeepAlive: 30 * time.Second,
+			}).DialContext,
+			ForceAttemptHTTP2:     true,
+			MaxIdleConns:          100,
```

Comment:
This looks correct to me as those are all just default parameters.
https://github.com/golang/go/blob/ac85f2bedd405d64b54c3a248c9079b6cde57359/src/net/http/transport.go#L50